src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js

```diff
@@ -0,0 +1,258 @@
+import React from 'react';
+import {
+    Dimensions,
+} from 'react-native';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+import {
+    deleteReportComment,
+} from '../../../../libs/actions/Report';
+import withLocalize, {withLocalizePropTypes} from '../../../../components/withLocalize';
+import PopoverWithMeasuredContent from '../../../../components/PopoverWithMeasuredContent';
+import ReportActionContextMenu from './ReportActionContextMenu';
+import ConfirmModal from '../../../../components/ConfirmModal';
+
+const propTypes = {
+    /** Update the callbacks on ContextMenu Context */
+    setValue: PropTypes.func.isRequired,
+    ...withLocalizePropTypes,
+};
+
+class PopoverReportActionContextMenu extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            reportID: 0,
+            reportAction: {},
+            selection: '',
+            reportActionDraftMessage: '',
+            isPopoverVisible: false,
+            isDeleteCommentConfirmModalVisible: false,
+            cursorRelativePosition: {
+                horizontal: 0,
+                vertical: 0,
+            },
+
+            // The horizontal and vertical position (relative to the screen) where the popover will display.
+            popoverAnchorPosition: {
+                horizontal: 0,
+                vertical: 0,
+            },
+        };
+        this.onPopoverHide = () => {};
+        this.contextMenuAchor = undefined;
+        this.showContextMenu = this.showContextMenu.bind(this);
+        this.hideContextMenu = this.hideContextMenu.bind(this);
+        this.measureContent = this.measureContent.bind(this);
+        this.measureContextMenuAnchorPosition = this.measureContextMenuAnchorPosition.bind(this);
+        this.confirmDeleteAndHideModal = this.confirmDeleteAndHideModal.bind(this);
+        this.hideDeleteConfirmModal = this.hideDeleteConfirmModal.bind(this);
+        this.showDeleteConfirmModal = this.showDeleteConfirmModal.bind(this);
+        this.contextMenuHidden = this.contextMenuHidden.bind(this);
+        this.getContextMenuMeasuredLocation = this.getContextMenuMeasuredLocation.bind(this);
+        this.isActiveReportAction = this.isActiveReportAction.bind(this);
+    }
+
+    componentDidMount() {
+        this.props.setValue({
+            showContextMenu: this.showContextMenu,
+            hideContextMenu: this.hideContextMenu,
+            isActiveReportAction: this.isActiveReportAction,
+            showDeleteConfirmModal: this.showDeleteConfirmModal,
+        });
```

Comment:
I find this is very confusing. But I want to suggest it's maybe not a good use of context because...

- The component only mounts once 
- It sets the value of the context - so it should maybe be a _provider_ (if anything)?
- But the fact that the values don't change begs the question of whether to use context at all

The value in having a context is that something will update it in the future and the consumers will automatically be updated. In this case, we are just setting some methods here so we can call them from other places.

It may be fine to share some methods with a consumer wrapped in the provider, but we should create them and pass them once instead of doing this prop callback that sets to state creating an immediate re-render?

Reply:
Yeah. I will drop context and use Callbacks. We can export all the required callbacks to an external lib/module and call them.
I will reply or act upon the other comments when this is done.